orc8r/cloud/go/obsidian/swagger/handlers/handlers.go

```diff
@@ -33,3 +73,105 @@
 		return c.String(http.StatusOK, combined)
 	}
 }
+
+// GetSpecHandler returns a routing handler which creates and serves the
+// raw YAML spec of a particular service.
+func GetSpecHandler(yamlCommon string) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		service, ok, err := getServiceName(c.Param("service"))
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+		if !ok {
+			return obsidian.HttpError(errors.New("service not found"), http.StatusNotFound)
+		}
+
+		combined, err := swagger.GetServiceSpec(yamlCommon, service)
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+
+		return c.String(http.StatusOK, combined)
+	}
+}
+
+// GetUIHandler returns a routing handler which serves the UI of a service.
+func GetUIHandler(tmpl *template.Template) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		service, ok, err := getServiceName(c.Param("service"))
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+		if !ok {
+			return obsidian.HttpError(errors.New("service not found"), http.StatusNotFound)
+		}
+
+		var buf bytes.Buffer
+		err = tmpl.Execute(&buf, UIInfo{URL: obsidian.StaticURLPrefix + "/v1/spec/" + service})
+		if err != nil {
+			return obsidian.HttpError(err, http.StatusInternalServerError)
+		}
+
+		return c.HTML(http.StatusOK, buf.String())
+	}
+}
+
+// registerSpecHandlers registers routes for Swagger specs.
+func registerSpecHandlers(e *echo.Echo, trailSlashMiddleware echo.MiddlewareFunc) error {
+	yamlCommon, err := swagger.GetCommonSpec()
+	if err != nil {
+		return err
+	}
+
+	if obsidian.EnableDynamicSwaggerSpecs {
+		e.GET(obsidian.StaticURLPrefix+"/v1/spec/", GetCombinedSpecHandler(yamlCommon))
+	} else { // If dynamic Swagger spec is off, serve static spec
+		e.GET(obsidian.StaticURLPrefix+"/v1/spec/", func(c echo.Context) error {
+			return c.File(obsidian.StaticURLPrefix + "/v1/spec/swagger.yml")
+		})
+	}
+
+	e.GET(obsidian.StaticURLPrefix+"/v1/spec", nil, trailSlashMiddleware)
```

Comment:
Swap this line with the blank line above. Cause we wanna group the handlers for the monolithic endpoint, then group the (single) handler for the per-service endpoint